examples/policies/soul-bound-policy/soul-bound-policy-v1.pact

```diff
@@ -0,0 +1,120 @@
+(namespace (read-msg 'ns))
+
+(module soul-bound-policy-v1 GOVERNANCE
+
+  (defconst ADMIN-KS:string "marmalade-v2.marmalade-contract-admin")
+
+  (defcap GOVERNANCE ()
+    (enforce-guard ADMIN-KS))
+
+  (implements kip.token-policy-v2)
+  (use kip.token-policy-v2 [token-info])
+  (use marmalade-v2.policy-manager)
+  (use marmalade-v2.guard-policy-v1 [MINT-GUARD-MSG-KEY BURN-GUARD-MSG-KEY GUARD_SUCCESS])
+
+  (defschema mint-record 
+    account:string
+  )
+
+  (deftable records:{mint-record})
+
+  (defun has-guard-policy:bool (policies)
+    (> (length (filter (lambda (policy) (= policy marmalade-v2.guard-policy-v1)) policies)) 0))
+
+  (defun enforce-init:bool
+    ( token:object{token-info}
+    )
+    @doc "The function is run at `create-token` step of marmalade-v2.ledger."
+
+    (require-capability (INIT-CALL (at "id" token) (at "precision" token) (at "uri" token) soul-bound-policy-v1))
+
+    (enforce (= (at 'precision token) 0) "Precision must be 0 for soul-bound tokens")
+
+    (let* (
+        (policies (at 'policies token))
+
+        (mint-guard (try GUARD_SUCCESS (read-msg MINT-GUARD-MSG-KEY) ))
+        (burn-guard (try GUARD_SUCCESS (read-msg BURN-GUARD-MSG-KEY) ))
+      )
+      (enforce (has-guard-policy policies) "Guard policy is required for soul-bound tokens")
+
+      (enforce (!= mint-guard GUARD_SUCCESS) "Mint guard is required for soul-bound tokens")
+      (enforce (!= burn-guard GUARD_SUCCESS) "Burn guard is required for soul-bound tokens")
+    )
+    true
+  )
+
+  (defun enforce-mint:bool
+    ( token:object{token-info}
+      account:string
+      guard:guard
+      amount:decimal
+    )
+    (require-capability (MINT-CALL (at "id" token) account amount soul-bound-policy-v1))
+
+    (enforce (= amount 1.0) "Amount must be 1.0 for soul-bound tokens")
+
+    (let* ((mint-record (try { 'account: "" } (read records (at 'id token))))
```

Comment:
Similar to above comment, this can be typed like this: 
```
    (let* ((mint-record:object{mint-record} (try { 'account: "" } (read records (at 'id token))))
          (has-mint-record:bool (!= (at 'account mint-record) "")))
 ```

Reply:
I added the types! 😄 

To your other comment, yes but it will throw `Failure: Database exception: Insert: row found for key t:...` and that is not really readable without understanding what the code does, and users may not understand it while trying to mint.

Reply:
That is also true. I try to remove as much duplicate work in the contract, because the amount of executed code also directly relates to gas amount, and believe it's the ui dev's job to communicate to the users. I'll leave it to your design choice to decide :)